snowflake.go

```diff
@@ -123,23 +124,35 @@ func (dialector Dialector) BindVarTo(writer clause.Writer, stmt *gorm.Statement,
 	writer.WriteByte('?')
 }
 
-//no quotes, quotes cause everything needing quotes
 func (dialector Dialector) QuoteTo(writer clause.Writer, str string) {
-	// writer.WriteString(strings.ToLower(str))
+	re := regexp.MustCompile(`(\S+)\((.+?)\)`)
```

Comment:
Something like this would be a little safer: ([a-zA-Z|_]+)\((.+?)\)

All functions are strings and underscores: https://docs.snowflake.com/en/sql-reference/functions-all.html

This would help with situations like ((....))

Reply:
Test string: t_est_((stuff ) = 3 (()) should have 1 match

Reply:
Merged in suggestion